LPS-93339 portal-search-ranking-web: Adds error states

thektan opened the pull request:
- Adds the new pin icons (http://issues.liferay.com/browse/LPS-93401)
- Error states when server requests fails to easily retry (http://issues.liferay.com/browse/LPS-93339)

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

thektan commented:
@oliv-yu I added a key to the icon to get the icon to re-render on hover. The icon would re-render fine in storybook, but for production builds it was converting the `<use xlinkHref ../>` to `<path>` elements and not updating. Adding a `key` forces a re-render. https://github.com/BryanEngler/liferay-portal/pull/334/commits/949f5290e1dab6e6f8a0fb59b4e1b45769777260#diff-fe5bc70135bfb8da2abca2e7c2e8d509R26

I also changed `xlinkHref` to `href` since [it's deprecated now](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href).

ccohick commented:
ci:test:bundle:norebase

BryanEngler commented:
added to dev branch and pushed, thanks
